src/runtime_src/tools/xclbinutil/SectionAIEMetadata.cxx

```diff
@@ -26,8 +26,11 @@ namespace XUtil = XclBinUtilities;
 // ----------------------------------------------------------------------------
 SectionAIEMetadata::init SectionAIEMetadata::initializer;
 
-SectionAIEMetadata::init::init() {
-  registerSectionCtor(AIE_METADATA, "AIE_METADATA", "", false, false, boost::factory<SectionAIEMetadata*>()); 
+SectionAIEMetadata::init::init() 
+{
+  auto sectionInfo = std::make_unique<SectionInfo>(AIE_METADATA, "AIE_METADATA", boost::factory<SectionAIEMetadata*>());
```

Comment:
Why use boost::factory instead of boost::value_factory? The only function that currently uses this variable is Section::createSectionObjectOfKind and it returns a pointer. Might be better to update it to return the object rather than a object pointer

Reply:
For the use cases associated with the Section option, we really do wish to create a Section in the heap and not the stack.